test/jdk/java/lang/reflect/Generics/TestMissingTypeVariable.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @library /test/lib
+ * @bug 8337302
+ * @enablePreview
+ * @summary Tests that an exception is thrown if a type variable is not declared
+ */
+
+import jdk.test.lib.ByteCodeLoader;
+
+import java.lang.classfile.ClassFile;
+import java.lang.classfile.Signature;
+import java.lang.classfile.attribute.SignatureAttribute;
+import java.lang.constant.ClassDesc;
+import java.lang.reflect.AccessFlag;
+import java.lang.reflect.Type;
+
+public class TestMissingTypeVariable {
+
+    public static void main(String[] args) throws Exception {
```

Comment:
To make the test more understandable to casual readers, I suggest putting the corresponding source code, as much as possible, as a comment.

Reply:
An alternative approach could be to write a package-private or nested class in the same file:
```java
static class Generic<T> {
    T field;
}

// ...

var cf = ClassFile.of();
var generic = cf.parse(Path.of(System.getProperty("test.classes"), "TestMissingTypeVariable$Generic.class"));
var result = cf.transformClass(generic, ClassTransform.dropping(ce -> ce instanceof SignatureAttribute);
var missing = ByteCodeLoader.load("TestMissingTypeVariable$Generic", result);
// proceed
```

Would be more straightforward to readers.

Reply:
I added a comment, I think that makes it clearest.